### PR TITLE
[fx] fix merge_matmul tests making invalid torch.split calls

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #74728
* #74727
* __->__ #75019
* #76675

The merge_matmul tests currently call `torch.split` with a list of
`torch.fx.Attribute` objects which won't work once `torch.split` is moved into
C++ because the argument parser actually checks the list contains integers
before calling `__torch_function__`.

### CODE REVIEW

**Review by jamesr66a — APPROVED** [review]
Original author of this pass doesn't work here anymore but I'm rubber stamping

maybe cc @wushirong @jfix71 want to take a look as well

File: torch/fx/experimental/merge_matmul.py [hunk 1]
```diff
@@ -9,20 +9,27 @@
...
-        The first dimension of t.
+        List of matmul results for each input tensor.
     """
-    return t.shape[0]
+    splits = [x.shape[0] for x in inputs]
+
+    # When fx tracer is running, x.shape[0] will be torch.fx.Attribute but we
+    # need an int even when tracing
+    if isinstance(result, torch.fx.Proxy):
+        splits = [0] * len(inputs)
```
Comment (jfix71) [inline torch/fx/experimental/merge_matmul.py:30]: Tiny nit, marginally inefficient when tracing 🙂 
```suggestion
    # When fx tracer is running, x.shape[0] will be torch.fx.Attribute but we
    # need an int even when tracing
    if isinstance(result, torch.fx.Proxy):
        splits = [0] * len(inputs)
    else:
        splits = [x.shape[0] for x in inputs]
```

**Review by jfix71 — APPROVED** [review]
LGTM!

**Pull request closed without merging**